### PR TITLE
feat(observablehq):  Add parser offset positions

### PR DESCRIPTION
**Pull request by GordonSmith**
Rename some internal methods
Extend InspectorFactory to include ID

Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [x] The change has been fully tested:
  - [ ] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

### CODE REVIEW

File: packages/observablehq-compiler/src/compiler.ts [hunk 1]
```diff
@@ -128,86 +200,43 @@ function createImportVariable(name?: string, alias?: string) {
...
-        w.importDefine(parsed);
-    };
-    return retVal;
-}
-type ModuleFunc = Awaited<ReturnType<typeof createModule>>;
-
-async function createCell(node: ohq.Node, baseUrl: string) {
+// Cell  ---
+async function createCell(node: ohq.Node, options: CompileOptions) {
+    node.id = "" + node.id;
```
Comment (jeclrsg) [inline packages/observablehq-compiler/src/compiler.ts:205]: Should this type just be a string?

**Review by jeclrsg — CHANGES_REQUESTED** [review]
@GordonSmith Noticed a bit of commented code that I think had been relocated. Also a couple of questions for things I wasn't sure about

File: packages/observablehq-compiler/src/compiler.ts [hunk 2]
```diff
@@ -71,13 +82,62 @@ async function importNotebook(partial: string) {
...
+        if (parsed.injections.length) {
+            mod = mod.derive(parsed.injections, main);
+        }
+        variables.forEach(v => v(main, inspector));
+        importVariables.forEach(v => v(main, mod));
+        return mod;
+    };
+    retVal.importVariables = importVariables;
+    retVal.variables = variables;
+    retVal.dispose = () => {
```
[the remark this answers is not in the document]
Comment (GordonSmith) [reply packages/observablehq-compiler/src/compiler.ts:114]: Let me review these again for consistency, but yes you are probably correct...

File: packages/observablehq-compiler/src/compiler.ts (hunk 1 above)
Comment (GordonSmith) [reply packages/observablehq-compiler/src/compiler.ts]: This is no longer needed and removed, thx.

**Review by jeclrsg — APPROVED** [review]

**Pull request merged**